### DISCUSSION TITLE
Getting Duplicate character in Words

### DISCUSSION THREAD

**Discussion by dksfaldu** [pymupdf/PyMuPDF#1625] [Q&A]
I am getting duplicate entry of the words when read pdf using 
```

doc = fitz.open(pdf_file_path)
page = doc[0]
words = page.get_text("words")
```

For Word in PDF : STOCK
I am getting in word : SSSSTTTTOOOOCCCCKKKK
[d91ba75d-1646-456c-9f2f-f5d7e168bd63.pdf](https://github.com/pymupdf/PyMuPDF/files/8192725/d91ba75d-1646-456c-9f2f-f5d7e168bd63.pdf)

PyMuPDF Version : 1.19.4/1.19.5

**Comment by JorjMcKie** [comment]
Your PDF is coded like that.
Sorry.

**Comment by JorjMcKie** [comment]
Obviously the PDF creator tried to simulate BOLD with this.

**Comment by JorjMcKie** [comment]
You need additional logic to cope with this. To learn how to, look at the PyMuPDF module. The following CLI will produce proper - even layout-preserving - text output: ``python -m fitz gettext your.pdf``.

**Reply by dksfaldu** [reply]
Hi 
Can we get the word location as well with the python for the same word like pixel points.

**Reply by JorjMcKie** [reply]
You have a lot of options here:
* you have the rectangle for each word tuple `word = (x0, y0, x1, y1, "word", ...)`. Put `wrect = fitz.Rect(word[:4])`.
* Determine the coordinates of each single character of the word:
```python
chars = []  # dictionaries of each single word character
for b in page.get_text("rawdict", clip=wrect)["blocks"]:  # extract word again on a by char basis
    for l in b["lines"]:
        for s in l["spans"]:
            for c in s["chars"]:
                chars.append(c)
chars.sort(key=lambda b: b["bbox"][0])  # to be sure sort chars by ascending x0
# now loop through the word's characters and identify those that
# (1) overlap each other (sufficiently), and
# (2) are the same unicode
c0 = chars[0]  # 1st char dict
c0rect = fitz.Rect(c0["bbox"])  # rect of this char
newword = c0["c"]  # recovered real word - start with 1st char
for c in chars[1:]:
    crect = fitz.Rect(c["bbox"])  # rect of this char
    if c["c"] == c0["c"] and abs(crect & c0rect) / abs(crect) >= 0.8:  # 80% rect overlap!
        # skip this char
        continue
    newword += c["c"]  # add new unicode
    c0 = c
    c0rect= crect
```
Something like the above should recover the word you want ...
